display-brightness-ddcutil@themightydeity.github.com/extension.js

```diff
@@ -208,26 +225,97 @@ export default class DDCUtilBrightnessControlExtension extends Extension {
+        if (mode === "original") {
+            displays.forEach(display => {
+                display.slider.setHideOSD();
+                display.slider.changeValue(newValue);
+                display.slider.resetOSD();
+            });
+        } else if (mode === "experimental") {
+            const oldValue = mainMenuButton.getStoredSliders()[0].old_value;
+
+            if (oldValue === newValue)
+                return;
+            const increased = (newValue > oldValue)
+
+            if (increased) {
+                const increase = newValue - oldValue
+                const remaining = 100 - oldValue
+                const frac = increase / remaining
+                displays.forEach(display => {
+                    display.slider.setHideOSD();
+                    const oldValue = 100 * display.slider.ValueSlider.value;
+                    const remaining = 100 - oldValue
+                    const increase = frac * remaining
+                    display.slider.changeValue(oldValue + increase);
+                    display.slider.resetOSD();
+                });
+            } else {
+                const decrease = oldValue - newValue
+                const remaining = oldValue
+                const frac = decrease / remaining
+                displays.forEach(display => {
+                    display.slider.setHideOSD();
+                    const oldValue = 100 * display.slider.ValueSlider.value;
+                    const remaining = oldValue
+                    const decrease = frac * remaining
+                    display.slider.changeValue(oldValue - decrease);
+                    display.slider.resetOSD();
+                });
+            }
+
+            mainMenuButton.getStoredSliders()[0].old_value = newValue;
+        }
+        pause_sync = false
     }
 
     addSettingsItem() {
         const settingsItem = new PopupMenu.PopupMenuItem(_('Settings'));
         settingsItem.connect('activate', () => {
             this.openPreferences();
         });
-        mainMenuButton.addMenuItem(settingsItem, 1);
-
+        if (this.settings.get_int('button-location') === 0) {
+            mainMenuButton.addMenuItem(settingsItem, 1);
+        } else if (this.settings.get_boolean('show-sliders-in-submenu') && this.settings.get_boolean('show-all-slider')) {
+            mainMenuButton.getStoredSliders()[0].menu.addMenuItem(settingsItem)
+        }
         const reloadItem = new PopupMenu.PopupMenuItem(_('Reload'));
         reloadItem.connect('activate', event => {
             this.reloadExtension();
         });
-        mainMenuButton.addMenuItem(reloadItem, 2);
+        if (this.settings.get_boolean('show-sliders-in-submenu') && this.settings.get_boolean('show-all-slider')) {
+            mainMenuButton.getStoredSliders()[0].menu.addMenuItem(reloadItem)
+        } else if (this.settings.get_int('button-location') === 0) {
+            mainMenuButton.addMenuItem(reloadItem, 2);
+        }
     }
```

Comment:
Need to flip the condition check or else `mainMenuButton.getStoredSliders()[0]` becomes undefined when  `this.settings.get_boolean('show-sliders-in-submenu') && this.settings.get_boolean('show-all-slider')` is true when  `this.settings.get_int('button-location') === 0` is true as well. Should be like Line 305-309.

Remeber to check that the extension works when the main button is in the top bar instead system menu of once you have all the features ready. It is bit of a hassle but the extension should work with both ways.